src/app/centroidSwitch.js

```diff
@@ -0,0 +1,163 @@
+      },
+      switchLayersFromProject: function (ids) {
+          // summary:
+          //      decides whether to show the centroids or not
+          // ids
+          console.log('app.centroidSwitch::switchLayersFromProject', arguments);
+
+          if (!ids || ids.length === 1) {
+              this.enabled = false;
+              return;
+          }
+
+          this.enabled = true;
+      },
+      switchLayersFromExtent: function (extent) {
+          // summary:
+          //      decides whether to show the centroids or not
+          // extent: https://developers.arcgis.com/javascript/jsapi/map-amd.html#event-extent-change
+          console.log('app.centroidSwitch::switchLayersFromExtent', arguments);
+
+          if (!this.enabled) {
+              return;
+          }
+
+          // if they are just panning don't worry about it
+          // if the delta is null, it is the first load so
+          // we need to show the centroids
+          if (!extent.levelChange && extent.delta) {
+              return;
+          }
+
+          this._loadLayers().then(lang.hitch(this, function () {
+              if (this.override) {
+                  this.centroidLayer.setVisibility(false);
+                  this.explodedLayer.forEach(function (layer) {
+                      layer.setVisibility(false);
+                  });
+                  return;
+              }
+
+              this.centroidLayer.setVisibility(extent.lod.level < this.scaleTrigger);
+              Object.keys(this.explodedLayer).forEach(function (key) {
+                  var layer = this.explodedLayer[key];
+                  layer.setVisibility(extent.lod.level >= this.scaleTrigger);
+              }, this);
+              // topic.publish(config.topics.layer.toggleVisibility, {
+              //     centroids: extent.lod.level < this.scaleTrigger,
+              //     projects: extent.lod.level >= this.scaleTrigger
+              // });
+          }));
+      },
+      _loadLayers: function () {
+          // summary:
+          //      adds the new layers if they haven't been yet
+          console.log('app.centroidSwitch::_loadLayers', arguments);
+
+          if (Object.keys(this.explodedLayer) === 0 || !this.centroidLayer) {
+              var li = config.layerIndices;
+              var deferreds = [];
+              var typesLookup = {
+                  0: 'poly-exploded',
+                  1: 'line-exploded',
+                  2: 'point-exploded'
+              };
+
+              [li.poly, li.line, li.point].forEach(function (layerIndex, i) {
+                  var layer = new FeatureLayer(config.urls.mapService + '/' + layerIndex, {
+                      id: typesLookup[i]
+                  });
+
+                  var deferred = new Deferred();
+
+                  layer.on('load', function () {
+                      deferred.resolve();
+                  });
```

Comment:
Being nit-picky. I think that this could be written as `layer.on('load', deferred.resolve);`.